src/nitsm/tsmcontext.py

```diff
@@ -1390,3 +1391,39 @@ def pins_to_custom_sessions(
             SemiconductorModuleContext._sessions[session_id] for session_id in session_ids
         )
         return pin_query_context, session_data, channel_group_ids, channel_lists
+
+    def publish_per_site(
+        self, measurements: "_PublishPerSiteMeasurementsArg", published_data_id="", pin=""
+    ) -> None:
+        """
+        Publishes measurements from multiple sites for the Semiconductor Multi Test step to consume.
+        Use this method when you want to publish data for multiple sites in the same order in which
+        the sites are defined in the Semiconductor Module context and you want to include the pin
+        name as well as the published_data_id.
```

Comment:
Although your text matches the help string in the .NET API, I think the .NET API should change to match the following:
```suggestion
        the sites are defined in the Semiconductor Module context.
```

Reply:
https://github.com/ni/nitsm-python/pull/144/commits/fae0edfc3a65be653c44128cb03041d6b706f539